utils/lock/cond_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package lock
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCond(t *testing.T) {
+	cancelled, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	log := func(args ...any) {
+		t.Log(args...)
+	}
```

Comment:
It appears that this function was moved to fix a data-race (based on the commit message).
1. Aren't we using the wrong logger now? The sub-tests are now logging into the parent test.
2. Why does using the parent testing struct not incur the same data-race as the child testing struct?
3. I'm surprised that the stdlib logger doesn't support concurrent logs... It isn't implemented on the interface...

Reply:
It actually supports concurrent calls, but the goroutines need to finish and not remain running once the test ends.

Fixed [here](https://github.com/ava-labs/avalanchego/pull/3999/commits/aa40197de464a9646a64b79fa21ff08b7d3b8db7#diff-21d4aef882132c8ba89d616ef779c42596a30aadefae9d8b66861b984ae8975fR112-R130).

And I don't know what I was thinking with the log function. I think I added a lock and somehow removed it by mistake.